frontend/components/AddressLink.tsx

```diff
@@ -4,18 +4,27 @@ import { EXPLORER_URL_BY_MIDDLEWARE_CHAIN } from '@/constants/urls';
 import { Address } from '@/types/Address';
 import { truncateAddress } from '@/utils/truncate';
 
-type AddressLinkProps = { address?: Address; hideLinkArrow?: boolean };
+type AddressLinkProps = {
+  address?: Address;
+  hideLinkArrow?: boolean;
+
+  // TODO: mark as required once balance breakdown is updated.
+  // and remove the default value
+  middlewareChain?: MiddlewareChain;
+};
 
 export const AddressLink = ({
   address,
   hideLinkArrow = false,
+  middlewareChain = MiddlewareChain.GNOSIS,
```

Comment:
should remove this fallback 
`= MiddlewareChain.GNOSIS,`

and only render if middlewareChain is passed